### ISSUE TITLE
Why is the ratio of mapped reads different between STAR and Qualimap2 using the same alignment file?

### ISSUE THREAD

**Issue by U201412486** [alexdobin/STAR#401]
Hi,I'm a green hand about STAR.I have a question. I need you help.
  when I use STAR to map,it produce this,

> ### Started job on |       Apr 14 18:02:19
>                              Started mapping on |       Apr 14 18:03:09
>                                     Finished on |       Apr 14 18:14:29
>        Mapping speed, Million of reads per hour |       74.67
> 
>                           Number of input reads |       14103798
>                       Average input read length |       194
>                                     UNIQUE READS:
>                    Uniquely mapped reads number |       12162042
>                         Uniquely mapped reads % |       86.23%
>                           Average mapped length |       192.41
>                        Number of splices: Total |       75922
>             Number of splices: Annotated (sjdb) |       25779
>                        Number of splices: GT/AG |       25733
>                        Number of splices: GC/AG |       761
>                        Number of splices: AT/AC |       237
>                Number of splices: Non-canonical |       49191
>                       Mismatch rate per base, % |       0.16%
>                          Deletion rate per base |       0.01%
>                         Deletion average length |       1.77
>                         Insertion rate per base |       0.00%
>                        Insertion average length |       1.52
 
**
But I use Qualimap2 (Evaluating next generation sequencing alignment data)for the same alignment file,it looks like this,  
      
![image](https://user-images.githubusercontent.com/35086008/38845951-82835438-41bf-11e8-885c-fecf2f16bf91.png)
  
Mapped reads % is not same above. How does STAR  caculate the ratio of mapped reads? What is the 
reason?
looking for you reply. thanks.

**Comment by alexdobin** [comment]
Hi @U201412486 

STAR counts paired-end reads as one read.
I think the Qualimap2 counts lines in the SAM file - if you did not output the unmapped reads, than 100% of reads in the SAM file are mapped.
Also, I think Qualimap2 counts all alignments of the multimapping reads, this is why the total number of lines in the SAM files is more than double of the total number of reads.

Cheers
Alex

**Comment by U201412486** [comment]
Hi @alexdobin 
     Thank you for you reply.
     You mean the ratio of mapped will be same between STAR and Qualimap2 if we use single-end reads,is it right? And will using BAM file have impact for the reason that you explain for?
     looking forward to you reply. thanks.

**Comment by alexdobin** [comment]
Hi @U201412486 

for single-end read there will be no factor of 2, so the numbers will be closer.
However, the issue with multimapping reads that Qualimap2 counts multiple times will remain.
Also, Qualimap2 will always report 100% mapping rate because the unmapped reads are not output into the BAM file by default - if you want to output unmapped reads, you need to use --outSAMunmapped Within option.
Basically, I would recommend that you use the numbers reported by STAR.

Cheers
Alex

**Comment by U201412486** [comment]
Thanks for your response!
